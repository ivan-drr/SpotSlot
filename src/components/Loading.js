import React, { Component } from 'react';
import '../styles/Loading.css';

class Loading extends Component {

  constructor(props) {
    super(props);
    this.typeWriter = this.typeWriter.bind(this);
  }

  typeWriter(str, speed, id, iterator, intervals) {
    if (document.getElementById(id) !== null) {
      const element = document.getElementById(id);

      setTimeout(() => {
        if (iterator < str.length) {
          element.innerHTML += str.charAt(iterator);
          iterator++;

          setTimeout(this.typeWriter(str, speed, id, iterator), speed);

          return true;
        } else return false;
      }, 1000/speed*2);

    }
  }

  render() {
    let iterator = 0;

    if (this.props._isFetch === 'error') {
      setTimeout(() => {
        this.typeWriter(' âœ˜ Error connecting to API', 70, 'errorText', iterator);
      }, 0);
      return (
        <div id="loading" className="text-danger fixed-top">
          <span id="errorText" className="ml-3"></span>
        </div>
      );
    } else if (!this.props._isFetch) {
      setTimeout(() => {
        this.typeWriter(' Loading...', 70, 'loadingText', iterator);
      }, 0);
      return(
        <div id="loading" className="text-primary fixed-top">
          <div className="spinner-border text-primary" role="status"></div>
          <span id="loadingText" className="ml-3"></span>
        </div>
      );
    } else return null;
  }

}

export default Loading;
