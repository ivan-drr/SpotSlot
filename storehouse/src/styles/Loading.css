#loading {
  margin-top: 8%;
}
