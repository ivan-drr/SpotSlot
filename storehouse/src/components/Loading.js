import React, { Component } from 'react';
import '../styles/Loading.css';

class Loading extends Component {

  constructor(props) {
    super(props);
    this.typeWriter = this.typeWriter.bind(this);
  }

  typeWriter(str, speed, id, iterator) {
    if (document.getElementById(id) !== null) {
      var element = document.getElementById(id);

      setTimeout(() => {
        if (iterator < str.length) {
          element.innerHTML += str.charAt(iterator);
          iterator++;
          setTimeout(this.typeWriter(str, speed, id, iterator), speed);
        }
      }, 1000/speed*2);
    }
  }

  render() {
    var iterator = 0;

    if (this.props._isFetch === 'error') {
      document.getElementById('loadingText').innerHTML = '';
      this.typeWriter(' âœ˜ Error connecting to API', 70, 'loadingText', iterator);
      return (
        <div id="loading" className="text-danger fixed-top">
          <span id="loadingText" className="ml-3"></span>
        </div>
      );

    } else if (!this.props._isFetch) {
      this.typeWriter(' Loading...', 70, 'loadingText', iterator);
      return(
        <div id="loading" className="text-primary fixed-top">

          <div className="spinner-border text-primary" role="status"></div>
          <span id="loadingText" className="ml-3"></span>
        </div>
      );
    } else return null;
  }

}

export default Loading;
