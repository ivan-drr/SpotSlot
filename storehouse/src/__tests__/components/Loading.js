import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Loading from '../../components/Loading';
import $ from 'jquery'

describe('Unit Tests', () => {
  describe('Render', function () {
    const loading = shallow(<Loading/>);

    it('should render correctly', () => {
      expect(loading).toMatchSnapshot();
    });

    it('should exist <div id="loading" className="text-primary fixed-top">', function () {

    });

    it('should fadeOut <span id="loadingText"> after fetch data', function () {

    });

    describe('typeWriter', function () {
      it('should write on the DOM element', function () {
        
      });

      it('should warn that the DOM element was not found', function () {

      });

      it('should stop writting on the DOM element after str.length iteration', function () {

      });
    });
  });

});
