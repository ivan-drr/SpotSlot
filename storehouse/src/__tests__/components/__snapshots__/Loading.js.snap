// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests Render should render correctly 1`] = `ShallowWrapper {}`;
